builtin/providers/aws/resource_aws_db_instance.go

```diff
@@ -271,6 +271,7 @@ func resourceAwsDbInstanceCreate(d *schema.ResourceData, meta interface{}) error
 			CopyTagsToSnapshot:         aws.Bool(d.Get("copy_tags_to_snapshot").(bool)),
 			DBInstanceClass:            aws.String(d.Get("instance_class").(string)),
 			DBInstanceIdentifier:       aws.String(d.Get("identifier").(string)),
+			DBSubnetGroupName:          aws.String(d.Get("db_subnet_group_name").(string)),
```

Comment:
For what it's worth, I removed this line in e52af33db7b2d03a34adb560eb4e37535a9a91d3 . It should only be conditionally set, as you've done below in 292-295